### PR TITLE
add tests on ruby 3.1, 3.2, rails 7.0, drop support for rails 5.1

### PR DESCRIPTION
**Pull request by razumau**
Adds tests on Ruby 3.1 and 3.2 and Rails 7.0. Drops support for Rails 5.1.

[Since Rails 7, datetime columns have precision 6 by default](https://github.com/rails/rails/pull/42297). I set the same value explicitly in the test database and switched time comparisons in several tests to rounding timestamps with `.to_i`.

### CODE REVIEW

File: .ruby-version [hunk 1]
```diff
@@ -0,0 +1 @@
+3.2.2
```
Comment (bquorning) [inline .ruby-version:1]: I think we set .ruby-version to the lowest compatible version, not the highest.

File: test/queue_audit_test.rb [hunk 2]
```diff
@@ -57,7 +57,7 @@ class QueueAuditTest < Minitest::Test
           Timecop.freeze(Time.now) do
             @audit.complete!
 
-            assert_equal Time.now, @audit.completed_at
+            assert_equal Time.now.to_i, @audit.completed_at.to_i
```
Comment (bquorning) [inline test/queue_audit_test.rb:60]: Why do we set the precision to 6 digits in the schema and then use 0 digits when we compare timestamps?

Comment (razumau) [reply test/queue_audit_test.rb:60]: These were separate changes: I set precision to 6 for consistency across tests, and then decided that `to_i`’s precision is good enough for these tests. Do you think we should test these more precisely?

Comment (bquorning) [reply test/queue_audit_test.rb:60]: The `to_i` conversion should not be necessary, especially inside a `Timecop.freeze` block. If the `to_i` is necessary for making _some_ of the tests pass, I suspect those tests fail once in a while when rounding `3.999997` to 3 and `4.000002` to 4.

Comment (razumau) [reply test/queue_audit_test.rb]: Reworked it to use `Time#floor`. Maybe there’s a better way to do this? The problem is that while in the database we now only have precision 6, `Time.now` still has additional nanoseconds, so we have to either cut them off or make a comparison less strict.

**Review by bquorning — APPROVED** [review]

**Pull request merged**